tests/samplers_tests/test_nsgaii.py

```diff
@@ -504,13 +504,14 @@ def test_crowding_distance_sort(values: List[List[float]]) -> None:
 def test_study_system_attr_for_population_cache() -> None:
     sampler = NSGAIISampler(population_size=10)
     study = optuna.create_study(directions=["minimize"], sampler=sampler)
+    study_system_attrs = study._storage.get_study_system_attrs(study._study_id)
```

Comment:
Currently, this test case depends on the implementation detail of `InMemoryStorage` (e.g. This test will be broken when using `RDBStorage`). How about moving this line into `get_cached_entries` function?